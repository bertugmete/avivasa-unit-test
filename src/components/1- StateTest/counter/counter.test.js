import React from 'react'
import { shallow } from 'enzyme'
import Counter from './index'
import renderer from 'react-test-renderer'

describe('Counter component', () => {
  it('Matches the snapshot', () => {
    const tree = renderer.create(<Counter />).toJSON()

    expect(tree).toMatchSnapshot()
  })
  it('counter state starts with 0', () => {
    const wrapper = shallow(<Counter />)
    expect(wrapper.find('p').text()).toEqual('Counter 0')
    expect(wrapper.state().count).toBe(0)
  })

  it('increment Button On Click', () => {
    const wrapper = shallow(<Counter />)
    const incrementButton = wrapper.find('button.increment')
    incrementButton.simulate('click')
    const text = wrapper.find('p').text()
    expect(text).toEqual('Counter 1')
  })

  it('decrement Button On Click', () => {
    const wrapper = shallow(<Counter />)
    const decrementButton = wrapper.find('button.decrement')
    decrementButton.simulate('click')
    const text = wrapper.find('p').text()
    expect(text).toEqual('Counter -1')
  })
})
