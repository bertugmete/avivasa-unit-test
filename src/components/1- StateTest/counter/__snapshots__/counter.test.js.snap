// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component Matches the snapshot 1`] = `
<div>
  <p>
    Counter 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
